LocalPackages/DataBrokerProtection/Sources/DataBrokerProtection/Scheduler/DataBrokerProtectionQueueManager.swift

```diff
@@ -191,24 +193,29 @@ private extension DefaultDataBrokerProtectionQueueManager {
                       priorityDate: mode.priorityDate,
                       showWebView: showWebView,
                       operationDependencies: operationDependencies,
+                      errorHandler: errorHandler,
                       completion: completion)
     }
 
     func cancelCurrentModeAndResetIfNeeded() {
         switch mode {
-        case .immediate(let completion), .scheduled(let completion):
+        case .immediate(let errorHandler, let completion), .scheduled(let errorHandler, let completion):
             operationQueue.cancelAllOperations()
             let errorCollection = DataBrokerProtectionAgentErrorCollection(oneTimeError: DataBrokerProtectionQueueError.interrupted, operationErrors: operationErrorsForCurrentOperations())
-            completion?(errorCollection)
-            resetModeAndClearErrors()
+            errorHandler?(errorCollection)
+            resetMode(clearErrors: true)
+            completion?()
+            resetMode()
```

Comment:
Why do we need to reset the mode a final time? If another operation runs as part of the completion, won’t that reset it as parts of it’s completion handler? (and so forth…and if there is no completion, then the initial ` resetMode(clearErrors: true)` would have reset it anyway).

Reply:
@aataraxiaa I did it so that the current behaviour (mode being reset after completion block) doesn't change and cause any unintended regression. In my testing, indeed reseting the mode once is enough for this particular code path to work, but I'm not sure about other existing flows 🤔

Reply:
Got it. It might be enough to just reset it once and then ensure out tests cover the known code path. However, I understand your reasoning, and setting it after completion does seem like the right thing to do. Not blocking, good to merge as is IMO.

Reply:
Hi sorry I missed this because I was out. Ideally I'd prefer it if we just had the one instance of reset, I don't think there's any particular need to preserve the existing behavior, but if we don't, can we add a comment explaining this? I think it's likely to be very confusing to a new person reading it

Reply:
@THISISDINOSAUR Good point. I'd address it in my current project.